### PR TITLE
DOC: Minor doc formatting.

### PR DESCRIPTION
**Pull request by Carreau**
- use continuation prompt `...` for better parsing of example.
- Use double backticks when snippet is not supposed to be a reference.
- Mode a dot that ends a sentence outside of reference backticks.
- Use `Examples` section to detect examples.
- Properly indent "Returns" to be picked by numpydoc.

### CODE REVIEW

File: scipy/sparse/linalg/dsolve/linsolve.py [hunk 1]
```diff
@@ -520,7 +520,7 @@ def spsolve_triangular(A, b, lower=True, overwrite_A=False, overwrite_b=False,
     Returns
     -------
     x : (M,) or (M, N) ndarray
-        Solution to the system `A x = b`. Shape of return matches shape of `b`.
+        Solution to the system ``A x = b``. Shape of return matches shape of `b`.
```
Comment (WarrenWeckesser) [inline scipy/sparse/linalg/dsolve/linsolve.py:523]: The modified line exceeds 79 characters.
```suggestion
        Solution to the system ``A x = b``. Shape of return matches shape
        of `b`.
```

Comment (Carreau) [reply scipy/sparse/linalg/dsolve/linsolve.py:523]: Gha.

**Review by WarrenWeckesser — APPROVED** [review]
Looks good, thanks!

**Comment by WarrenWeckesser** [comment]
Thanks @Carreau, merged.  (Test failures were unrelated to this PR.)

**Comment by Carreau** [comment]
Thanks!

**Pull request merged**